### PR TITLE
Feature/uweb 5310 hero contract

### PR DESCRIPTION
**Pull request by evdtempel**
Interfaces in TypeScript cause issues, because they don’t actually “exist”. They are only syntactic sugar for the compiler, i.e. you can’t do `instanceof` on an interface.

There is two choices we can make: “Hungarian notation” or defining a type-property. Ed, Alex and I had a quick discussion about it last friday and settled on the “Hungarian notation”.

Let’s have a face-to-face discussion about it with the team, not through Slack or PR-comments.

### CODE REVIEW

File: src/atoms/ICtaContentLink.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import {ICta} from "./ICta";
+
+export class ICtaContentLink extends ICta {}
```
Comment (zwaardje) [inline src/atoms/ICtaContentLink.ts]: Why add a new type if there is no difference

File: src/organisms/IHero.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import {ICtaBlock} from "../molecules/ICtaBlock";
+import {IImage} from "../atoms/IImage";
+
+export abstract class IHero {
+    // HEAD
```
Comment (zwaardje) [inline src/organisms/IHero.ts]: Please remove comments, not needed. Code should be self-explanatory.

File: src/atoms/IButton.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import {ICta} from "./ICta";
+
+export class IButton extends ICta {
```
Comment (zwaardje) [inline src/atoms/IButton.ts]: As discussed in the previous meetings. We are using interfaces and not classes. The interface then are used across the different projects.

File: src/atoms/IButtonWithIcon.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import {IButton} from "./IButton";
+
+export class IButtonWithIcon extends IButton {
```
Comment (zwaardje) [inline src/atoms/IButtonWithIcon.ts]: @rhelling @alphen87  are we going to use the I<Name> for the <Name>Interface.

Are we still going to use the enum modifier to add classes to the "button" (for example).

File: src/organisms/IHeroBasic.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import {IHero} from "./IHero";
+
+export class IHeroBasic extends IHero {
```
Comment (zwaardje) [inline src/organisms/IHeroBasic.ts]: Where are we able to differentiate the type? We should use and an enum to contain the type of organisms we currently create. It should be added to each organism to differentiate between them.

File: src/atoms/IButtonWithIcon.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+    icon: string;
```
Comment (zwaardje) [inline src/atoms/IButtonWithIcon.ts]: Add an interface for icon. It is going to be used on a storybook level. We could also add an enum list of iconTypes to get even stricter.

**Comment by zwaardje** [comment]
We should also add tslint which extends the airbnb tslint config to meet the Ultimaker.com code quality guidelines: https://github.com/Ultimaker/Meta

File: src/atoms/IButtonWithIcon.ts (hunk 4 above)
Comment (alphen87) [reply src/atoms/IButtonWithIcon.ts]: Please use ButtonInterface

File: src/atoms/IButton.ts (hunk 3 above)
Comment (alphen87) [reply src/atoms/IButton.ts]: I agree, interfaces please

File: src/atoms/IButtonWithIcon.ts (hunk 4 above)
Comment (evdtempel) [reply src/atoms/IButtonWithIcon.ts]: Interface suffixes as discussed

File: src/organisms/hero/HeroBasic.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import {Hero} from "./Hero";
+
+export class HeroBasic extends Hero {
+    readonly paragraph: string;
```
Comment (rhelling) [inline src/organisms/hero/HeroBasic.ts]: This is a common hero property, can you move it to the base Hero. We could have multiple types of paragraphs in future, so we call this one description at this moment.

Comment (evdtempel) [reply src/organisms/hero/HeroBasic.ts]: Fair point, renamed and moved!

**Pull request merged**